public/app/features/explore/Logs/LogsTable.tsx

```diff
@@ -141,25 +166,93 @@ export const LogsTable: React.FunctionComponent<Props> = (props) => {
           },
         });
       });
+
```

Comment:
This is a lot of work inside an effect. Do you think we can extract this to a couple of functions that we can call from here?

Reply:
Created some helper functions to clean this up